### DISCUSSION TITLE
[Magick++] Composite/edit jobs on image made using appendImages only use first image

### DISCUSSION THREAD

**Discussion by TheEssem** [ImageMagick/ImageMagick#2538] [Help] [answered]
I'm attempting to run an edit job on an image generated with the Magick++ `appendImages` function. The image in question can be found here: https://cdn.discordapp.com/attachments/363882035049267209/753337399769301042/caption.png
![image](https://user-images.githubusercontent.com/21112968/92653099-7a1cd880-f2b3-11ea-9947-210da3bade1d.png)

The "test" text and thumbs up image were separate images appended together. When I try to run the edit job on the image, I get this: https://cdn.discordapp.com/attachments/363882035049267209/753337424188276756/motivate.png
![image](https://user-images.githubusercontent.com/21112968/92653122-82751380-f2b3-11ea-9d8f-13030c156d4e.png)

Another example: https://cdn.discordapp.com/attachments/363882035049267209/753337848861818952/globe.gif
![globe](https://user-images.githubusercontent.com/21112968/92653222-a7698680-f2b3-11ea-9dd9-f413dd474d03.gif)

As you can see, it only seems to grab the top part of the image. What do I do? Here's my code for appending the image: https://github.com/esmBot/esmBot/blob/0d8e14ed90a1b0edde8ed997eb9f90e4faa7af05/natives/caption.cc#L14

And the code for the edit jobs: https://github.com/esmBot/esmBot/blob/0d8e14ed90a1b0edde8ed997eb9f90e4faa7af05/natives/motivate.cc#L14 https://github.com/esmBot/esmBot/blob/0d8e14ed90a1b0edde8ed997eb9f90e4faa7af05/natives/globe.cc#L14

**Comment by fmw42** [comment, marked as answer]
Add the equivalent of +repage after reading your input image(s)

**Reply by TheEssem** [reply]
`image.repage()` did the trick, thanks!

**Comment by snibgo** [comment]
As Fred says, more or less. When I download and identify caption.png, I get:
```
caption2.png PNG 72x89 72x17+0+0 16-bit sRGB 2624B 0.000u 0:00.015
```
Can you see the problem? You need the equivalent of "+repage" after the append.
